projects/plugins/boost/app/modules/cache/Page_Cache.php

```diff
@@ -57,9 +57,10 @@ private function create_advanced_cache() {
 				return new \WP_Error( 'advanced-cache.php exists but is not the correct file' );
 			}
 		} else {
-			$contents = '<?php
+			$plugin_name = basename( dirname( plugin_dir_path( __FILE__ ), 3 ) );
+			$contents    = '<?php
 // Boost Cache Plugin 0.1
-require_once( ABSPATH . \'/wp-content/plugins/boost/app/modules/cache/Boost_File_Cache.php\' );
+require_once( ABSPATH . \'/wp-content/plugins/' . $plugin_name . '/app/modules/cache/Boost_File_Cache.php\' );
```

Comment:
Looking at this code, I think this is going to become a really annoying corner in the system, fraught with lots of edge-cases: 
- Users can configure their site to use a different `wp-content` directory by setting `WP_CONTENT`.
- They can also override their plugins directory using `WP_PLUGINS`.
- Who knows what else they will do to their site 😄 

First up, if something goes wrong with this setup, the user's site will crash. Would it make sense to either use `include_once`, or if that might produce PHP warnings, check the target file exists first? That way if we get this step wrong we won't break sites.

Second, do you think it's worth using absolute paths here? Instead of trying to guess at parent directories, we could just use `dirname( plugin_dir_path( __FILE__ ), 3 )` without calling `basename` to get the absolute path to the Boost plugin and use a path relative to that?

Of course, that has the slight disadvantage that caching will break if a user moves their site to a different directory. In theory, we could account for that by auto-detecting how much of the path matches `ABSPATH` and stripping it off and using a path relative to that -- but honestly, I think that may be more complex than is worth it. 

What do you think?

Reply:
I forgot to push another fix I had for this that checks if the Boost_File_Cache.php exists, sorry. I guess that takes care of if they move the plugin to a different place?

In WPSC I use register_shutdown_function() to display an error in the html source, but we could define a constant that's checked by the module and then emails the user, or shows an admin_notice. I don't think we need to do that now though.

Reply:
I tried using plugins/$plugin_name but I guess since advanced-cache.php is loaded by settings.php then it needs wp-content/ in it, so we might as well use WP_CONTENT_DIR.
WP_PLUGIN_DIR is not available yet unfortunately. It's loaded later in settings.php

Thanks Peter for suggestions. :)

Reply:
Would it be safe to resolve the boost dir while writing the file and hardcoding while writing it in `advance-cache.php`?

Reply:
> Would it be safe to resolve the boost dir while writing the file and hardcoding while writing it in advance-cache.php?

Yeah, I think putting the absolute path in `advanced-cache.php` is likely the way to go. But to double-check make sure the file exists before requiring it and using it.

Reply:
I hard coded the path. There is a file_exists() check there too, so it should be robust enough to not mess things up if Boost is moved, or even if the plugins directory is moved.
In a later version we should pass on errors to the plugin so it can fix the file, but that's an edge case.